Source/Core/Common/UPnP.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_UPNP
+
+#include "Common/UPnP.h"
+
+#include "Common/Logging/Log.h"
+
+#include <array>
+#include <cstdlib>
+#include <cstring>
+#include <miniupnpc.h>
+#include <miniwget.h>
+#include <string>
+#include <thread>
+#include <upnpcommands.h>
+#include <vector>
+
+static struct UPNPUrls s_urls;
```

Comment:
Since you're cleaning up stuff, would you also mind removing `struct` here? It shouldn't be needed here and below